### PR TITLE
Changed http code on create index API with bad input raising NotXContentException to 400

### PR DESCRIPTION
**Pull request by ayushKataria**
### Description
[Describe what this change achieves]
Changed http code on create index API with bad input raising NotXContentException to 400

### Issues Resolved
[List any issues this PR will resolve]
#2756

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — CHANGES_REQUESTED** [review]
This needs a test, please.

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Fix weighted routing metadata deserialization error on process restart ([#4691](https://github.com/opensearch-project/OpenSearch/pull/4691))
 - Refactor Base Action class javadocs to OpenSearch.API ([#4732](https://github.com/opensearch-project/OpenSearch/pull/4732))
 - Migrate client transports to Apache HttpClient / Core 5.x ([#4459](https://github.com/opensearch-project/OpenSearch/pull/4459))
+- Changed http code on create index API with bad input raising NotXContentException to 400([#4773](https://github.com/opensearch-project/OpenSearch/pull/4773))
```
[the remark this answers is not in the document]
Comment (ayushKataria) [reply CHANGELOG.md]: done

**Comment by ayushKataria** [comment]
@dblock Added the test as suggested. Not sure what the tests failing are. Can you help with this?

File: CHANGELOG.md [hunk 2]
```diff
@@ -91,6 +91,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Fix weighted routing metadata deserialization error on process restart ([#4691](https://github.com/opensearch-project/OpenSearch/pull/4691))
 - Refactor Base Action class javadocs to OpenSearch.API ([#4732](https://github.com/opensearch-project/OpenSearch/pull/4732))
 - Migrate client transports to Apache HttpClient / Core 5.x ([#4459](https://github.com/opensearch-project/OpenSearch/pull/4459))
+- Changed http code on create index API with bad input raising NotXContentException to 400 ([#4773](https://github.com/opensearch-project/OpenSearch/pull/4773))
```
Comment (dblock) [inline CHANGELOG.md:94]: What was the old code (say changed from X to 400)

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
I wrote a quick guide to how to deal with flakey tests in https://github.com/opensearch-project/OpenSearch/pull/4868/files - care to try it for me?

**Review by mch2 — APPROVED** [review]
Thanks for taking this @ayushKataria.

**Comment by mch2** [comment]
Retriggered check - unrelated failure.
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.shard.SegmentReplicationIndexShardTests.testReplicaReceivesLowerGeneration" -Dtests.seed=25A1BC6ABB78D54B -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=hu -Dtests.timezone=Australia/Darwin -Druntime.java=17

org.opensearch.index.shard.SegmentReplicationIndexShardTests > testReplicaReceivesLowerGeneration FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([25A1BC6ABB78D54B:AACF30236089C3AF]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.index.shard.SegmentReplicationIndexShardTests.assertEqualCommittedSegments(SegmentReplicationIndexShardTests.java:717)
        at org.opensearch.index.shard.SegmentReplicationIndexShardTests.testReplicaReceivesLowerGeneration(SegmentReplicationIndexShardTests.java:306)
```

File: CHANGELOG.md (hunk 2 above)
Comment (ayushKataria) [reply CHANGELOG.md]: Done

**Comment by dblock** [comment]
@ayushKataria rebase this? lets see if the tests are still flaky, then just work through them

**Comment by ayushKataria** [comment]
@dblock I was seeing some build errors earlier, but when rebase updated this today it has gone through successfully

File: CHANGELOG.md [hunk 3]
```diff
@@ -42,6 +42,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Relax visibility of the HTTP_CHANNEL_KEY and HTTP_SERVER_CHANNEL_KEY to make it possible for the plugins to access associated Netty4HttpChannel / Netty4HttpServerChannel instance ([#4638](https://github.com/opensearch-project/OpenSearch/pull/4638))
 - Use ReplicationFailedException instead of OpensearchException in ReplicationTarget ([#4725](https://github.com/opensearch-project/OpenSearch/pull/4725))
 - Migrate client transports to Apache HttpClient / Core 5.x ([#4459](https://github.com/opensearch-project/OpenSearch/pull/4459))
+- Changed http code on create index API with bad input raising NotXContentException from 500 to 400 ([#4773](https://github.com/opensearch-project/OpenSearch/pull/4773))
```
Comment (dblock) [inline CHANGELOG.md:45]: There's a bunch of changes here that don't belong to this PR.

**Review by dblock — CHANGES_REQUESTED** [review]
See below for CHANGELOG:

File: CHANGELOG.md (hunk 3 above)
Comment (ayushKataria) [reply CHANGELOG.md:45]: Weird. They seem to have come in when I synced the fork. I have removed them from this PR now

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
DCO and precommit are failing :( back to you @ayushKataria lmk if you need help (I recommend you squash these commits on your local and sign the 1 commit left with `git commit -s`).

**Comment by dblock** [comment]
@ayushKataria do you need help with DCO?

**Comment by ayushKataria** [comment]
@dblock Yes, I would appreciate some help with this. What happened is that for the change in CHANGELOG, I was travelling so I pushed from a different machine but missed the sign off on that. I tried squashing the commits, but since I have merged from upstream, it says "has 2 parents", so, I am not able to squash and push with a single signed off commit. What would you suggest I should do now?

For the gradle check, I am not sure what the issue is. It seems to be failing at a build stage and not a test failure. This is the task it seems to be failing at
`Execution failed for task ':distribution:bwc:staged:buildBwcLinuxTar'.
> Building 2.4.0 didn't generate expected file /var/jenkins/workspace/gradle-check/search/distribution/bwc/staged/build/bwc/checkout-2.4/distribution/archives/linux-tar/build/distributions/opensearch-min-2.4.0-SNAPSHOT-linux-x64.tar.gz`

**Comment by dblock** [comment]
I think the easiest is to do a subset of https://code.dblock.org/2015/08/31/getting-out-of-your-first-git-mess.html, ie. create a new clean branch, merge your changes onto it, then force push that clean slate to the `xcontentstatuscode` branch on GitHub to update this PR.

```
git checkout main 
git pull upstream main 
git checkout -b xcontentstatuscode-squashed 
git merge xcontentstatuscode --squash
# examine the diff, ensure it's all nice and clean
git add .
git commit -s -m "...."
git push origin xcontentstatuscode-squashed:xcontentstatuscode -f
```

**Comment by ayushKataria** [comment]
@dblock Thanks, it's gone through now

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Thanks for hanging in here with us @ayushKataria! Merged.

@CEHENKLE @andrross We don't backport this one to 2.x or do we? :)
